tests/functional/test_config_groups.py

```diff
@@ -110,51 +132,128 @@ def test_that_groups_not_allowed_to_intersect_in_cluster(self, sdk_client_fs, cl
-            )
-        self.assert_that_host_add_is_unavailable(component_group_2, test_host_1)
-        self.assert_host_candidate_equal_expected(component_group_2.host_candidate(), 1)
+            component_group_2 = service.component(name=FIRST_COMPONENT_NAME).group_config_create(name=SECOND_GROUP)
+        assert_that_host_add_is_unavailable(component_group_2, test_host_1)
+        assert_host_candidate_equal_expected(component_group_2.host_candidate(), [SECOND_HOST])
+
+
+class TestIncorrectHostInGroups:
+    @pytest.fixture()
+    def cluster_with_components_on_one_host(
+        self, create_two_hosts, cluster: Cluster, provider: Provider
+    ) -> Tuple[Service, Host, Host]:
+        """Add service, two hosts and create components to check config groups"""
+
+        service = cluster.service_add(name='test_service_1')
+        test_host_1, test_host_2 = create_two_hosts
+        cluster.host_add(test_host_1)
+        cluster.hostcomponent_set(
+            (test_host_1, service.component(name=FIRST_COMPONENT_NAME)),
+            (test_host_1, service.component(name=SECOND_COMPONENT_NAME)),
+        )
+        return service, test_host_1, test_host_2
+
+    def test_add_incorrect_host_to_provider_group(self, sdk_client_fs, provider_bundle, provider):
+        """Test exception rise when we try to add incorrect host to provider group"""
+        with allure.step("Create host from first provider"):
+            correct_host = provider.host_create(fqdn=utils.random_string())
+        with allure.step("Create second provider"):
+            provider_2 = provider_bundle.provider_create(name="Second test provider")
+        with allure.step("Create host from second provider"):
+            incorrect_host = provider_2.host_create(fqdn=utils.random_string())
+        with allure.step("Create config group for first provider and try to add the first host"):
+            provider_group = provider.group_config_create(name=incorrect_host.fqdn)
+            assert_that_host_add_is_unavailable(provider_group, incorrect_host)
+            assert_host_candidate_equal_expected(provider_group.host_candidate(), [correct_host.fqdn])
+        with allure.step("Add first host to provider group and check that second add is not available"):
+            provider_group.host_add(correct_host)
+            assert_that_host_exists(provider_group, correct_host)
+            assert_host_candidate_equal_expected(provider_group.host_candidate(), [])
+
+    def test_add_incorrect_host_to_service_group(self, cluster_with_components_on_one_host):
+        """Test exception rise when we try to add incorrect host to service group"""
+
+        service, test_host_1, test_host_2 = cluster_with_components_on_one_host
+        with allure.step("Create group for service"):
+            service_group = service.group_config_create(name=FIRST_GROUP)
+        with allure.step("Create the second group for service and check that not allowed to add the first host to it"):
```

Comment:
this is step from previous tests, please check all and fix

Reply:
fixed. thnx!